app/models/popular_post_builder.rb

```diff
@@ -0,0 +1,42 @@
+class PopularPostBuilder
+
+  class << self
+    def build(site)
+      new(site).build
+    end
+  end
+
+  def initialize(site)
+    @site = site
+  end
+
+  def build
+    api_response = client.popular_posts(view_id: @site.analytics_viewid, dimention: @site.ranking_dimension)
+    ranking = []
+    api_response[:reports].first[:data][:rows].each do |row|
+      public_id = row[:dimensions].first.match(/\d+/).to_s.to_i
+      older_value = row[:metrics].first[:values].first.to_f
```

Comment:
Why don't you wrap response via client?
Client should create response object instead of pure Hash I think.